src/services/ContributingDeveloperAuditService/scm/GitHub/api/GitHubApiClient.ts

```diff
@@ -0,0 +1,215 @@
+import axios, { AxiosInstance, AxiosResponse } from "axios";
+import { soosLogger } from "../../../../../logging/SOOSLogger";
+import { sleep } from "../../../../../utilities";
+
+interface IHttpRequestParameters {
+  baseUri: string;
+  githubPAT: string;
+}
+
+interface IHttpClientParameters extends IHttpRequestParameters {
+  apiClientName: string;
+}
+
+export interface GitHubOrganization {
+  login: string;
+}
+
+export interface GitHubRepository {
+  id: number;
+  name: string;
+  full_name: string;
+  private: boolean;
+  owner: GitHubOrganization;
+}
+
+export interface ContributingDeveloper {
+  username: string;
+  repositories: ContributingDeveloperRepositories[];
+}
+
+interface ContributingDeveloperRepositories {
+  id: number;
+  name: string;
+  lastCommit: string;
+  isPrivate: boolean;
+}
+
+export interface Commits {
+  commit: {
+    author: Author;
+  };
+}
+
+export interface Author {
+  name: string;
+  email: string;
+  date: string;
+}
+
+export const threeMonthsDate = (() => {
```

Comment:
make this a "days ago" function so it it more useful - it can be in the root utilities functions like StringUtilties, e.g. DateUtilities and a method e.g.
```suggestion
const DateUtilities = {
  getDate: (
    daysAgo: number
  ): Date => {
    
  }
```